ui-core/src/components/inputs/datePicker/DatePicker.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import cx from 'classnames';
+import { Calendar as CalendarIcon } from '@osrd-project/ui-icons';
+import Input, { InputProps } from '../Input';
+import CalendarPicker, { CalendarPickerProps } from './CalendarPicker';
+import useDatePicker from './useDatePicker';
+
+export type DatePickerProps = {
+  inputProps: InputProps;
+  calendarPickerProps: Omit<CalendarPickerProps, 'modalPosition' | 'calendarPickerRef'>;
+};
```

Comment:
I think you can change the type of DatePickerProps like this. This way, the onDateChange handle either a date or a calendarSlot, it will be easier to use the component in osrd or any other application
```suggestion
export type DatePickerProps = (
  | {
      isRangeMode?: false;
      onDateChange: (nextDate: Date | null | undefined) => void;
    }
  | {
      isRangeMode: true;
      onDateChange: (nextSelectedSlot: CalendarSlot | undefined) => void;
    }
) & {
  inputProps: InputProps;
  calendarPickerProps: Omit<CalendarPickerProps, 'modalPosition' | 'calendarPickerRef'>;
};
```

You will have to remove isRangeMode and onDateChange from CalendarPickerProps